source/multisource.go

```diff
@@ -39,10 +39,12 @@ func (ms *multiSource) Endpoints(ctx context.Context) ([]*endpoint.Endpoint, err
 		}
 		if len(ms.defaultTargets) > 0 {
 			for i := range endpoints {
-				endpoints[i].Targets = ms.defaultTargets
+				eps := endpointsForHostname(endpoints[i].DNSName, ms.defaultTargets, endpoints[i].RecordTTL, endpoints[i].ProviderSpecific, endpoints[i].SetIdentifier)
```

Comment:
This strips the `Labels`. Is that intentional?

Reply:
I think it was ignorance. Have added it and have adjusted the tests.